module-os/board/rt1051/_exit.cpp

```diff
@@ -38,10 +38,18 @@
 #include <stdbool.h>
 #include <string.h>
 #include <exit_backtrace.h>
+#include <purefs/vfs_subsystem.hpp>
 
 
 static void __attribute__((noreturn)) stop_system(void)
 {
+    const auto err = purefs::subsystem::unmount_all();
+    if(err) {
+        LOG_WARN("Unable unmount all filesystems with error: %i.", err);
+    } else {
+        LOG_INFO("Filesystems unmounted successfully...");
+    }
+    LOG_INFO("Restarting the system...");
```

Comment:
_exit could be also called on normal system shutdown (or should be :smile:  ), not only on restart.

Reply:
In our design system is always rebooted in case of the exit